symengine/expression.h

```diff
@@ -32,16 +32,16 @@ class Expression
     //! Construct Expression from `int`
     Expression(int n) : m_basic(integer(n)) {}
     //! Construct Expression from Basic
-    template <typename T, typename = typename std::enable_if<std::is_constructible<RCP<const Basic>, T &&>::value>::type>
+    template <typename T>
```

Comment:
@certik some pieces of piranha do. Most complex stuff triggers internal compiler errors. I tried with some 15.x version, if I recall correctly. ``noexcept`` was most definitely supported (I never had to write macros for that and I use ``noexcept`` in every move constructor).

For a while I implemented various workarounds in an effort to support ICC, but eventually it became too much of  a burden. I got far enough that ``fateman1_perf`` would compile before giving up, but I think after a few recent changes it will be broken again.